File: src/main/java/lec04/Lec04Main.java
package lec04;

public class Lec04Main {

    public static void main(String[] args) {
        JavaMoney money1 = new JavaMoney(2000L);
        JavaMoney money2 = new JavaMoney(1000L);
        JavaMoney money3 = money1;
        JavaMoney money4 = new JavaMoney(2000L);

        if (money1.compareTo(money2) > 0) {
            System.out.println("Money1이 Money2보다 금액이 큽니다.");
        }

        System.out.println(money1 == money3);
        System.out.println(money1.equals(money4));

        JavaMoney plussedMoney = money2.plus(money4);
        System.out.println("plussedMoney = " + plussedMoney);
    }
}

File: src/main/kotlin/lec04/Lec04Main.kt
package lec04

fun main() {
    val money1 = JavaMoney(2000L)
    val money2 = JavaMoney(1000L)
    val money3 = money1
    val money4 = JavaMoney(2000L)

    if (money1 > money2) {
        println("Money1이 Money2보다 금액이 큽니다.")
    }
    println(money1 === money3)
    println(money1 == money4)

    // Lazy 연산
    if (fun1() || fun2()) {
        println("본문")
    }

    val money5 = Money(1000L)
    val money6 = Money(2000L)
    println(money5.plus(money6))
    println(money5 + money6)
}


fun fun1(): Boolean {
    println("fun1")
    return true
}

fun fun2(): Boolean {
    println("fun2")
    return true
}